src/DateField.spec.ts

```diff
@@ -0,0 +1,104 @@
+
+    DateField({egress:"YY/MM/DD"})
+
+    // known strings?
+    DateField({egressShorthand:"iso"})
+    DateField({egress:"iso-with-tz"})
+
+    //what about parsing options
+    DateField({egressShorthand:"iso"})
+    DateField({egress:"iso-with-tz"})
+
+    // what about a strict option
+    DateField({parseOptions: {strict:true}})
+    DateField({parseOptions: {parseString: "YY/MM/DD"}}) // and reject other formats?
+    
+    expect(1).toBe(1)
+  })
+  
+
+
+  test('DateCast handles null cases', () => {
+    // eventually I want this 2500 test file from pandas to be our goal for DWIM date parsing
+    // https://github.com/pandas-dev/pandas/blob/main/pandas/tests/tools/test_to_datetime.py
+    assertDC(undefined, null)
+    assertDC(null, null)
+    assertDC('', null)
+  })
+
+  test('DateCast handles real dates', () => {
+    const dString = '2022-07-30'
+    const d = new Date(dString)
+    //assertDC(dString, d)  // this is the original datecast test, commented out because it doesn't work, the following does work
+
+    const dString2 = '2022-07-30T16:00:00.000Z'
+    const d2 = new Date(dString2)
+    assertDC(dString2, d2)
+  })
+
+  test('DateCast handles invalid dates as errors', () => {  
+    assertThrow(
+      '2022-07-35',
+      "'2022-07-35' parsed to 'null' which is invalid"
+    )
+  })
+  
+  test('DateCast handles random strings as errors', () => {  
+    assertThrow('foo', "'foo' parsed to 'null' which is invalid")
+  })
+
+  test('DateCast handles infinity as an error', () => {  
+    assertThrow(1 / 0, "'Infinity' parsed to 'null' which is invalid")
+
+  })
+
+  test('instantiate DateField', () => {
+    const d = DateField()
+    console.log(d)
+    expect(1).toBe(1)
+  })
+  test('pandas date functions', () => {
+    assertDC("02/17/2009", new Date('2009-02-17T17:00:00.000Z')) //"MM/DD/YY" #Month-Day-Year with leading zeros
+    assertDC("18/02/2009", new Date('2009-02-18T17:00:00.000Z')) //"DD/MM/YY" #Day-Month-Year with leading zeros (
+    assertDC("2009/02/19", new Date('2009-02-19T17:00:00.000Z')) //"YY/MM/DD" #Year-Month-Day with leading zeros
+    assertDC("February 20, 2009", new Date('2009-02-20T17:00:00.000Z')) //"Month D, Yr" Month name-Day-Year with no leading zeros
+    assertDC("2/21/2009", new Date('2009-02-21T17:00:00.000Z')) //"M/D/YY"#Month-Day-Year with no leading zeros
+    assertDC("22/2/2009", new Date('2009-02-22T17:00:00.000Z')), //"D/M/YY"#Day-Month-Year with no leading zeros
+    assertDC("2009/2/23", new Date('2009-02-23T17:00:00.000Z')), //"YY/M/D"#Year-Month-Day with no leading zeros
+    assertDC(" 2/24/2009", new Date('2009-02-24T17:00:00.000Z')) //"bM/bD/YY" #Month-Day-Year with spaces instead of leading zeros
+    assertDC("25Feb2009", new Date('2009-02-25T17:00:00.000Z')) // "DDMonYY" #Day-Month abbreviation-Year with leading zeros
+
+  })
+  */
+  test('js dropin replacement', () => {
+    expect(ChronoDateCast("02/17/2009")).toStrictEqual(new Date("02/17/2009"))
+  })
+
```

Comment:
commented out a bunch of tests, trying to get chrono.js to behave with same timezone handling as native js Date (not sure if that's the best idea, maybe we should default to GMT or NYC time, check what pandas does).
But for now trying to figure out how to have exact same behavior as JS Date

this fails locally with the following error 
```
  ● Cast Function tests -> › js dropin replacement

    expect(received).toStrictEqual(expected) // deep equality

    Expected: 2009-02-17T05:00:00.000Z
    Received: 2009-02-17T17:00:00.000Z
```

Reply:
After another pairing session with @hansjhoffman some notes

1. JS Dates default to current timezone at noon
2. ChronoJS dates default to current timezone at 00:00 
3. pandas timestamp defaults to GMT at 00
4. we want default behaviour of GMT because it is more predicatable, code won't give a different result based on where its run
5. we will expose as an option to emulate JS default behavior
6. our api won't mimic chronos.js
7. our new field will be called `SmartDateField`